adapters/revcontent/revcontent.go

```diff
@@ -0,0 +1,111 @@
+package revcontent
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+)
+
+type Adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the Revcontent adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	bidder := &Adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *Adapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	reqBody, err := json.Marshal(request)
+
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	if err := checkRequest(request); err != nil {
+		return nil, []error{err}
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+
+	req := &adapters.RequestData{
+		Method:  "POST",
+		Uri:     a.endpoint,
+		Body:    reqBody,
+		Headers: headers,
+	}
+	return []*adapters.RequestData{req}, nil
+}
+
+func checkRequest(request *openrtb.BidRequest) error {
+	if (request.App == nil || len(request.App.Name) == 0) && (request.Site == nil || len(request.Site.Domain) == 0) {
+		return &errortypes.BadInput{
+			Message: "Impression is missing app name or site domain, and must contain one.",
+		}
+	}
+
+	if request.App != nil && request.Site != nil {
```

Comment:
This check is performed by the "core" and isn't necessary here. See: https://github.com/prebid/prebid-server/blob/a709baa4f842b11d4711b642c68ee245639dde07/endpoints/openrtb2/auction.go#L336